lib/mongoose_push/api/v1.ex

```diff
@@ -32,20 +32,14 @@ defmodule MongoosePush.API.V1 do
           |> Map.delete(:device_id)
           |> transform_alert()
 
-        case MongoosePush.push(device_id, notification) do
-          :ok ->
-            conn
-            |> put_status(200)
-            |> json(nil)
-          {:error, reason} when is_atom(reason) ->
-            conn
-            |> put_status(500)
-            |> json(%{:details => reason})
-          {:error, _reason} ->
-            conn
-            |> put_status(500)
-            |> json(nil)
-        end
+        {status, payload} =
+          device_id
+          |> MongoosePush.push(notification)
+          |> MongoosePush.API.Errors.to_status
```

Comment:
Couldn't you match on the error here? IMO `Errors` module shouldn't handle the success case. Or, you could rename the function, something along the lines of `push_result_to_response`?

Reply:
Ofc such function could be moved to `MongoosePush.API` module